osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -327,6 +330,12 @@ public virtual void Save(BeatmapInfo beatmapInfo, IBeatmap beatmapContent, ISkin
 
                     setInfo.CopyChangesToRealm(liveBeatmapSet);
 
+                    foreach (var collection in r.All<BeatmapCollection>())
+                    {
+                        if (collection.BeatmapMD5Hashes.Remove(oldMd5Hash))
+                            collection.BeatmapMD5Hashes.Add(beatmapInfo.MD5Hash);
+                    }
```

Comment:
Does this loop need to try and touch every single collection in the DB? Is there no way to only query for the collections that actually contain the hash that needs updating on the DB side?

Reply:
It doesn't seem possible - trying to use `.Where(c => c.BeatmapMD5Hashes.Contains(oldMd5Hash))`, it errors with "The method Contains is not supported", and the same with `.Any(itm => itm == oldMd5Hash))`

Also, it seems like any errors in this function cause the entire beatmap data to be lost, which is a little bit scary.

Reply:
I think using the same structure as

https://github.com/ppy/osu/blob/3eda284b03b5abcabe78b03d743078313f8f52a0/osu.Game/Beatmaps/BeatmapImporter.cs#L142-L156

would be beneficial, potentially moving the code to a common location. Or probably making the `static transferCollectionReferences` more `public` or `internal` and using it in the place you are.